core/src/edu/cornell/gdiac/main/CollisionController.java

```diff
@@ -127,8 +114,37 @@ public void endContact(Contact contact) {
    */
   public void preSolve(Contact contact, Manifold oldManifold) {
     // Pre-solve collision handling
+    Fixture fix1 = contact.getFixtureA();
+    Fixture fix2 = contact.getFixtureB();
+    Body body1 = fix1.getBody();
+    Body body2 = fix2.getBody();
+    PlayerModel ply=level.getAvatar();
+    try {
+      Obstacle bd1 = (Obstacle) body1.getUserData();
+      Obstacle bd2 = (Obstacle) body2.getUserData();
+    if (bd1.equals(ply)){
+      if( bd1 instanceof BouncePlatformModel){
+      BouncePlatformModel bplt=(BouncePlatformModel) bd2;
+      float c= bplt.getCoefficient();
+      if (ply.isFrozen()){
+        contact.setRestitution(c);
+      }}
+    }
+    if (bd2.equals(ply)){
+      if( bd1 instanceof BouncePlatformModel){
+      BouncePlatformModel bplt=(BouncePlatformModel) bd1;
+      float c= bplt.getCoefficient();
+      if (ply.isFrozen()){
+        contact.setRestitution(c);
+        }
+      }
+    }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```

Comment:
could you abstract this to a separate method in collision controller just for code organization? Maybe you could call it `preSolveBounce` or something.